src/Core/Geographic/Extent.js

```diff
@@ -296,14 +296,31 @@ Extent.prototype.offsetScale = function offsetScale(bbox) {
  * @param {type} bbox
  * @returns {Boolean}
  */
-Extent.prototype.intersect = function intersect(bbox) {
+Extent.prototype.intersects = function intersect(bbox) {
     const other = bbox.as(this.crs());
     return !(this.west() >= other.east(this._internalStorageUnit) ||
              this.east() <= other.west(this._internalStorageUnit) ||
              this.south() >= other.north(this._internalStorageUnit) ||
              this.north() <= other.south(this._internalStorageUnit));
 };
 
+/**
+ * @documentation: Return the intersection of this extent with another one
+ * @param {type} other
+ * @returns {Boolean}
+ */
+Extent.prototype.intersect = function intersect(other) {
+    if (!this.intersects(other)) {
+        return new Extent(this.crs(), 0, 0, 0, 0);
+    }
+    return new Extent(this.crs(),
+        Math.max(this.west(), other.west(this._internalStorageUnit)),
+        Math.min(this.east(), other.east(this._internalStorageUnit)),
+        Math.max(this.south(), other.south(this._internalStorageUnit)),
+        Math.min(this.north(), other.north(this._internalStorageUnit)));
```

Comment:
I'm wondering if there are some funny CRS where the axis would go the other way ? (where west would be the max, and east the min)

Reply:
FWIW leaflet fixed a similar bug https://github.com/Leaflet/Leaflet/issues/4886, and this CRS https://epsg.io/9843-method 

These could be just a curiosity though. Also, this might not be the place here to support such CRS